website/views/shared/tasks/task.jade

```diff
@@ -44,7 +44,7 @@ li(bindonce='list', id='task-{{::task.id}}', ng-repeat='task in obj[list.type+"s
           span.glyphicon.glyphicon-bullhorn(tooltip=env.t('challenge'))
           | &nbsp;
       // delete
-      a(ng-if='!task.challenge.id', ng-click='removeTask(obj[list.type+"s"], $index)', tooltip=env.t('delete'))
```

Comment:
Yep - method was made here: https://github.com/HabitRPG/habitrpg/commit/e45d8307e713bb2b7732beb59567f4fa2cd102f2#diff-7a6d770c287cf03133942144e9f2c670R45

It looks like at one point, the method was used to both delete the task, as well as modify the UI-bound list array. so the list needed to be passed in. That's no longer the case, but the method signature was just never changed.